Windows - Adding background and for color to mulititextinput

Spitskool42 opened the pull request:
Adding background and foreground color support to multilinetextinput.

This makes it possible to change colors. Only color that is still fixed is the placeholder text color (SystemColors.GrayText). Tried to make placeholder text transparent, but this is not supported by winforms.richtextbox.

<!--- Describe your changes in detail -->
<!--- What problem does this change solve? -->
<!--- If this PR relates to an issue, include Refs #XXX or Fixes #XXX -->

## PR Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

freakboy3742 reviewed (changes requested):
Looks like it makes sense; however, as I indicated [in my review comments on #1535](https://github.com/beeware/toga/pull/1535#pullrequestreview-1054922854), a key part of every PR is the testing regiment. You need to (in order of preference) either:
1. Tell us which existing example demonstrates the behavior being implemented
2. Modify an existing example to add a demonstration of this behavior
3. Add a new example that demonstrates the behavior 
4. Provide code for a sample app that demonstrates the behavior
5. Describe the testing regimen

(5) is generally only acceptable when the feature being added is very difficult to demonstrate in a meaningful way; that generally won't apply to feature additions like this one.

Spitskool42 commented:
@freakboy3742 I added a testing program that I created with cookiecutter. Is this done correctly? 
I created this program to test colors on different widgets, I also want to use this test program later, to test colors in android (in another pull request: #1535).

I was able to execute and test this program on windows 10. But for some reason I couldn't get it started on android.  "The system cannot find the path specified."
` C:\Users\Gebruiker\AppData\Local\BeeWare\briefcase\Cache\tools\android_sdk\cmdline-tools\latest\bin\sdkmanager.bat --licenses`.
This is correct because I do not have a AppData\Local\Beeware\ folder yet. I do not have this problem on the official briefcase 0.3.7 release that I use for apps atm. I think it is not working on the briefcase master atm for android.

Spitskool42 commented:
@freakboy3742  thanks for the reply.

Added 10 buttons now, and changed variable and function names.
Only downside is that the foreground color Transparent is not allowed by default, this is why I disabled this button, maybe someone later on can implement this functionality.

I am not sure if we can 'reset' colors, I think we only set color when the user sets the color. And by default no color is set on most widgets which translates to default system coloring. I set the reset colors now to black and transparent (windows default I think).

freakboy3742 commented:
> I am not sure if we can 'reset' colors, I think we only set color when the user sets the color. And by default no color is set on most widgets which translates to default system coloring.

That's what I meant by reset - `color = None`. That isn't handled on every platform... but that's part of the problem :-)

freakboy3742 commented on examples/colors/colors/app.py:
```diff
@@ -0,0 +1,181 @@
+                toga.Label('Background color selection:'),
+                self.color_background_selection,
+                toga.Label(' '),
+                button_changecollor
```
These "flexible" color options aren't really needed any more.

freakboy3742 commented on examples/colors/colors/app.py:
```diff
@@ -0,0 +1,181 @@
+        control_box = toga.Box(
+            children=[
+                toga.Label('Color selection:'),
+                b_change_fcollor_r,
```
Some spelling issues here

freakboy3742 reviewed (approved):
Thanks for those updates; to expedite progress, I've made a few tweaks (including cleaning up the design issues that I flagged inline). 

Thanks for the contribution - this will be very helpful in testing color changes.
